build/chip_build.gni

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2020 Project CHIP Authors
```

Comment:
the directory /build/ is in the .gitignore file... could we move these from build/ to somewhere else? That we we don't conflict with the existing autotools stuff.

Reply:
We can, however the convention that GN recommends and the large GN projects follow is that definitions that are broadly relevant are placed inside //build. We may come across build files for third party code we want to reuse that expect to import definitions from //build. In those cases following the convention could be the difference between being able to reuse the build file as-is after adding compatible definitions at the expected path, and having to make local changes to the build file.

Currently there's nothing that will break from rearranging, but if we do it we should be aware that we're increasing the cost of reuse. We may want to move the artifacts directory instead, since I think that has no long term downside.

Reply:
the .gitignore references /build/ - won't these conflict? I'd love for these two systems to co-exist for a while, before we cut over to just gn.

Reply:
It seems like a bigger change to both introduce GN and rename build outputs at the same time. Could we rearrange now and then maybe move back to //build in a separate change, once we decide of an updated output name that throughout?

This is already a very large change - the less we touch the better.

Reply:
Seems fine to me, just want to make sure it doesn't break the existing stuff!

Reply:
Sure, will move them. Any suggestions on where? gn/ ?

Reply:
I fixed using secondary_source (see `gn help dotfile`) which allows us to put it at //gn/build while calling it //build. Therefore third_party build files that look for things at //build would still work, at the cost of a bit of inconsistency about how paths work (until we undo it). Very easy to undo in that case, just a git mv.